pkg/apis/batch/types.go

```diff
@@ -138,6 +138,27 @@ const (
 	PodFailurePolicyOnExitCodesOpNotIn PodFailurePolicyOnExitCodesOperator = "NotIn"
 )
 
+// This field controls when we recreate pods
+// When PodFailurePolicy is turned off we allow TerminatingOrFailed or Failed.
+// If PodFailurePolicy is on we only allow Failed.
+// The API interprets nil as default.
+// If PodFailurePolicy is set then nil is interpreted to be Failed.
+// If PodFailurePolicy is not set then nill is
+// interpreted to be TerminatingOrFailed.
+// +enum
+type PodReplacementPolicy string
+
+const (
+	// TerminatingOrFailed means that we recreate pods as soon as
+	// DeletionTimestamp is non nil.
+	TerminatingOrFailed PodReplacementPolicy = "TerminatingOrFailed"
+	// Failed means to recreate pods when they are marked as failed.
+	// This is the only allowed value when PodFailurePolicy is enabled.
```

Comment:
I don't see this part "This is the only allowed value when PodFailurePolicy" needed here. It is explained at the field definition.